playwright/fixtures/index.ts

```diff
@@ -35,113 +44,242 @@ export type Fixtures = {
+        if (!prismicUrl) {
+          await page.route(
+            "*/**/_manager",
+            async (route) => {
+              const postDataBuffer = route.request().postDataBuffer() as Buffer;
+              const postData = decode(postDataBuffer) as Record<
+                "procedurePath",
+                unknown[]
+              >;
+
+              if (postData.procedurePath[0] === "getState") {
+                const response = await route.fetch();
+                const existingBody = await response.body();
+                const existingData = (
+                  decode(existingBody) as Record<"data", unknown>
+                ).data as Record<
+                  "env",
+                  {
+                    endpoints: { PrismicWroom: string };
+                  }
+                >;
+
+                // Get Prismic URL from the response of the getState call
+                prismicUrl = existingData.env.endpoints.PrismicWroom;
 
-    const page = await context.newPage();
+                await route.continue();
+              }
+            },
+            // Ensure only the first getState call is intercepted
+            {
+              times: 1,
+            },
+          );
 
-    // Use the fixture value in the test.
-    await use(page);
+          // Visit the page to trigger the getState call and wait for it
+          await page.goto("/");
+          await page.waitForResponse("*/**/_manager");
+        }
 
-    // Gracefully close up everything
-    await page.close();
-    await context.close();
-  },
-});
+        const activeEnv = prismicUrl?.includes("wroom") ? "WROOM" : "PRISMIC";
+        const email = process.env[`${activeEnv}_EMAIL`];
+        const password = process.env[`${activeEnv}_PASSWORD`];
+
+        if (!prismicUrl) {
+          console.warn("Could not find Prismic URL.");
+        } else if (!email || !password) {
+          console.warn(
+            `Missing EMAIL or PASSWORD environment variables for ${activeEnv} environment.`,
+          );
+        } else {
+          // Do the authentication call
+          const res = await fetch(
+            new URL("./authentication/signin", prismicUrl).toString(),
+            {
+              method: "post",
+              body: JSON.stringify({
+                email,
+                password,
+              }),
+              headers: {
+                "Content-Type": "application/json",
+              },
+            },
+          );
+
+          if (!res.headers.has("Set-Cookie")) {
+            // If the authentication fails, log the error
+            const reason = await res.text();
+            console.error(
+              "Could not authenticate to prismic. Please check the credentials.",
+              reason,
+            );
+          } else {
+            // If the authentication succeeded, save the cookies to persist it
+            await fs.writeFile(
+              path.join(os.homedir(), ".prismic"),
```

Comment:
Same question as above: are we writing to the actual `.prismic` file on the test runner's machine? It would be great if we could mock this somehow so we aren't overwriting our own logins.

Reply:
Yes it will override it. For me it should work like this. I don't see a better option. Mocking the login kind of create an integration test on that part and can be tricky later on to handle for e2e tests. You think it's a problem to override your .prismic while you doing test?